UnityProject/Assets/Scripts/SpawnPointList.cs

```diff
@@ -73,16 +91,63 @@ public IEnumerator MoveContentToPosition(GameObject selectedObj)
       var startPosition = new Vector2(listContentTransform.anchoredPosition.x, listContentTransform.anchoredPosition.y);
       var time = 0f;
 
-      while (time < _movementDuration)
+      while (time < _effectDuration)
       {
-        ContentTransform.anchoredPosition = Vector2.Lerp(startPosition, targetPosition, time / _movementDuration);
+        ContentTransform.anchoredPosition = Vector2.Lerp(startPosition, targetPosition, 1 - Mathf.Pow(1 - time / _effectDuration, 3));
         time += Time.deltaTime;
         yield return null;
       }
 
       ContentTransform.anchoredPosition = targetPosition;
     }
 
+    public IEnumerator UpdateItemAlphas(int selectedObjIndex)
+    {
+      var time = 0f;
+      var childCount = ContentTransform.childCount;
+      var startColors = new Color[childCount];
+      var targetColors = new Color[childCount];
+      var listItems = new SpawnPointListItem[childCount];
+
+      for (int i = 0; i < childCount; i++)
+      {
+        listItems[i] = ContentTransform.GetChild(i).GetComponent<SpawnPointListItem>();
+        startColors[i] = listItems[i].Text.color;
+        targetColors[i] = listItems[i].Text.color;
+        var distanceFromCenter = Math.Abs(selectedObjIndex - i);
+        var targetAlpha = distanceFromCenter == 0 ? 1 : Mathf.Max(0, Mathf.Min(1 - AlphaStep * distanceFromCenter, 1));
+        targetColors[i].a = targetAlpha;
+      }
+
+
+      while (time < _effectDuration)
+      {
+        for (int i = 0; i < childCount; i++)
+        {
+          listItems[i].Text.color = Color.Lerp(startColors[i], targetColors[i], time / _effectDuration);
+        }
+        time += Time.deltaTime;
+        yield return null;
+      }
+
+      for (int i = 0; i < childCount; i++)
+      {
+        listItems[i].Text.color = targetColors[i];
+      }
+    }
+
+    public void SetItemAlphas(int selectedObjIndex)
+    {
+      for (int i = 0; i < ContentTransform.childCount; i++)
+      {
+        var text = ContentTransform.GetChild(i).GetComponent<SpawnPointListItem>().Text;
+        var color = text.color;
+        var distanceFromCenter = Math.Abs(selectedObjIndex - i);
+        color.a = distanceFromCenter == 0 ? 1 : Mathf.Max(0, Mathf.Min(1 - AlphaStep * distanceFromCenter, 1));
+        text.color = color;
+      }
+    }
+

```

Comment:
An easier way to do this would be to have each cell set its own alpha in its Update() (or in a method that the manager calls on it to tell it to update). That way we don't need to think about animating the effect over time when the list changes, since each cell is constantly updating its alpha to reflect its distance. I think an even more idiomatic way to represent this pattern is to make a component called alphaFalloff or something, that has a field for a transform that it uses as the point to reference when adjusting its alpha (in update again, you can require that the object its on has a canvas, text, sprite, etc and getComponent it)

Reply:
Done, thank you.